pkg/apimachinery/wait/wait.go

```diff
@@ -0,0 +1,39 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package wait
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// PollImmediately polls at fixed intervals without backoff, passing a context to the condition function.
+// This is useful if the condition function needs access to context (e.g., for cancellation).
+func PollImmediately(interval, timeout time.Duration, condition wait.ConditionWithContextFunc) error {
+	return wait.PollUntilContextTimeout(context.Background(), interval, timeout, true, condition)
+}
+
+// PollUntil polls with an increasing interval (backoff), passing a context to the condition function.
+// This is useful if the condition function needs access to context (e.g., for cancellation).
+func PollUntil(interval, timeout time.Duration, condition wait.ConditionWithContextFunc) error {
+	return wait.PollUntilContextTimeout(context.Background(), interval, timeout, false, condition)
+}
```

Comment:
It seems both function is thin wrapper to `PollUntilContextTimeout`, with one modification that is the `immediate` argument, is that correct?
If so, I would call these function directly instead of having these wrappers.

In addition, the functions suggest the different between the two is the rate-limiter behavior (`PollImmediately` - fixed interval, `PollUntil` - increasing interval), and as said above the difference seem to be in the `immediate` var.
According to the `k8s.io/apimachinery/pkg/util/wait` library godocs [1] it doesnt seem the `immediate` var to control the rate-limiting.
The `immediate` var, which control whether the given condition func will be executed before the initial way or after.

[1] https://github.com/kubernetes/apimachinery/blob/v0.30.0/pkg/util/wait/poll.go#L29-L32

Reply:
> It seems both function is thin wrapper to PollUntilContextTimeout, with one modification that is the immediate argument, is that correct?
> If so, I would call these function directly instead of having these wrappers.

There was a long review already and the reasoning for this wrappers have been outlined there.

> In addition, the functions suggest the different between the two is the rate-limiter behavior (PollImmediately - fixed interval, PollUntil - increasing interval), and as said above the difference seem to be in the immediate var.

I guess you refer to the function description.

Reply:
@RokibulHasan7 , can you look at the function text-doc again?

Reply:
function doc updated.

Reply:
Thanks, it looks good now.

Looking at the wrappers again, it seem `PollUntil` is not used at all.
Please consider remove it, it can be added later on upon necessity.

Reply:
Done!